libpod/container_internal_common.go

```diff
@@ -2432,6 +2426,46 @@ func (c *Container) generateCurrentUserPasswdEntry() (string, int, int, error) {
 	return pwd, uid, rootless.GetRootlessGID(), nil
 }
 
+// Sets the HOME Env variable with precedence: caller home, execUser home
+func (c *Container) setHomeEnvIfNeeded() error {
```

Comment:
if this is only used once, can it just be inline?

Reply:
I think a separate function is better.